PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:16 + add:0, change matches:
-    _inherit = 'account.move.line'
-    _inherit = "account.analytic.line"
-    _inherit = 'account.analytic.applicability'
-    _inherit = 'crm.team'
-    _inherit = 'payment.provider'
-    _inherit = 'payment.transaction'
-    _inherit = "product.category"
-    _inherit = 'product.document'
-    _inherit = 'product.product'
-    _inherit = "product.attribute.custom.value"
-    _inherit = 'product.packaging'
-    _inherit = 'product.template'
-    _inherit = 'res.company'
-    _inherit = 'res.partner'
-    _inherit = 'analytic.mixin'
-    _inherit = 'utm.campaign'

Total Changes: 39

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/sale/models/account_move.py
+++ b/addons/sale/models/account_move.py
@@ -6,7 +6,6 @@ from odoo.exceptions import UserError
 
 
 class AccountMove(models.Model):
-    _name = 'account.move'
     _inherit = ['account.move', 'utm.mixin']
 
     @api.model

--- a/addons/sale/models/account_move_line.py
+++ b/addons/sale/models/account_move_line.py
@@ -6,7 +6,7 @@ from odoo.tools import float_compare, float_is_zero
 
 
 class AccountMoveLine(models.Model):
-    _inherit = 'account.move.line'
+    _inherit = ['account.move.line']
 
     is_downpayment = fields.Boolean()
     sale_line_ids = fields.Many2many(

--- a/addons/sale/models/analytic.py
+++ b/addons/sale/models/analytic.py
@@ -5,13 +5,13 @@ from odoo import fields, models
 
 
 class AccountAnalyticLine(models.Model):
-    _inherit = "account.analytic.line"
+    _inherit = ["account.analytic.line"]
 
     so_line = fields.Many2one('sale.order.line', string='Sales Order Item', domain=[('qty_delivered_method', '=', 'analytic')], index='btree_not_null')
 
 
 class AccountAnalyticApplicability(models.Model):
-    _inherit = 'account.analytic.applicability'
+    _inherit = ['account.analytic.applicability']
     _description = "Analytic Plan's Applicabilities"
 
     business_domain = fields.Selection(

--- a/addons/sale/models/chart_template.py
+++ b/addons/sale/models/chart_template.py
@@ -2,7 +2,7 @@ from odoo import models
 
 
 class AccountChartTemplate(models.AbstractModel):
-    _inherit = 'account.chart.template'
+    _inherit = ['account.chart.template']
 
     def _get_property_accounts(self, additional_properties):
         property_accounts = super()._get_property_accounts(additional_properties)

--- a/addons/sale/models/crm_team.py
+++ b/addons/sale/models/crm_team.py
@@ -7,7 +7,7 @@ from odoo.tools import SQL
 
 
 class CrmTeam(models.Model):
-    _inherit = 'crm.team'
+    _inherit = ['crm.team']
 
     invoiced = fields.Float(
         compute='_compute_invoiced',

--- a/addons/sale/models/payment_provider.py
+++ b/addons/sale/models/payment_provider.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class PaymentProvider(models.Model):
-    _inherit = 'payment.provider'
+    _inherit = ['payment.provider']
 
     so_reference_type = fields.Selection(string='Communication',
         selection=[

--- a/addons/sale/models/payment_transaction.py
+++ b/addons/sale/models/payment_transaction.py
@@ -8,7 +8,7 @@ from odoo.tools import str2bool
 
 
 class PaymentTransaction(models.Model):
-    _inherit = 'payment.transaction'
+    _inherit = ['payment.transaction']
 
     sale_order_ids = fields.Many2many('sale.order', 'sale_order_transaction_rel', 'transaction_id', 'sale_order_id',
                                       string='Sales Orders', copy=False, readonly=True)

--- a/addons/sale/models/product_category.py
+++ b/addons/sale/models/product_category.py
@@ -2,7 +2,7 @@ from odoo import models, fields
 
 
 class ProductCategory(models.Model):
-    _inherit = "product.category"
+    _inherit = ["product.category"]
 
     property_account_downpayment_categ_id = fields.Many2one(
         comodel_name='account.account',

--- a/addons/sale/models/product_document.py
+++ b/addons/sale/models/product_document.py
@@ -4,7 +4,7 @@ from odoo import fields, models
 
 
 class ProductDocument(models.Model):
-    _inherit = 'product.document'
+    _inherit = ['product.document']
 
     attached_on_sale = fields.Selection(
         selection=[

--- a/addons/sale/models/product_product.py
+++ b/addons/sale/models/product_product.py
@@ -8,7 +8,7 @@ from odoo.tools import float_round
 
 
 class ProductProduct(models.Model):
-    _inherit = 'product.product'
+    _inherit = ['product.product']
 
     sales_count = fields.Float(compute='_compute_sales_count', string='Sold', digits='Product Unit of Measure')
 
@@ -100,7 +100,7 @@ class ProductProduct(models.Model):
 
 
 class ProductAttributeCustomValue(models.Model):
-    _inherit = "product.attribute.custom.value"
+    _inherit = ["product.attribute.custom.value"]
 
     sale_order_line_id = fields.Many2one('sale.order.line', string="Sales Order Line", ondelete='cascade')
 
@@ -108,7 +108,8 @@ class ProductAttributeCustomValue(models.Model):
         ('sol_custom_value_unique', 'unique(custom_product_template_attribute_value_id, sale_order_line_id)', "Only one Custom Value is allowed per Attribute Value per Sales Order Line.")
     ]
 
+
 class ProductPackaging(models.Model):
-    _inherit = 'product.packaging'
+    _inherit = ['product.packaging']
 
     sales = fields.Boolean("Sales", default=True, help="If true, the packaging can be used for sales orders")

--- a/addons/sale/models/product_template.py
+++ b/addons/sale/models/product_template.py
@@ -10,7 +10,7 @@ from odoo.addons.base.models.res_partner import WARNING_HELP, WARNING_MESSAGE
 
 
 class ProductTemplate(models.Model):
-    _inherit = 'product.template'
+    _inherit = ['product.template']
     _check_company_auto = True
 
     service_type = fields.Selection(

--- a/addons/sale/models/res_company.py
+++ b/addons/sale/models/res_company.py
@@ -5,7 +5,7 @@ from odoo.exceptions import ValidationError
 
 
 class ResCompany(models.Model):
-    _inherit = 'res.company'
+    _inherit = ['res.company']
     _check_company_auto = True
 
     _sql_constraints = [

--- a/addons/sale/models/res_partner.py
+++ b/addons/sale/models/res_partner.py
@@ -6,7 +6,7 @@ from odoo.osv import expression
 
 
 class ResPartner(models.Model):
-    _inherit = 'res.partner'
+    _inherit = ['res.partner']
 
     sale_order_count = fields.Integer(
         string="Sale Order Count",

--- a/addons/sale/models/sale_order.py
+++ b/addons/sale/models/sale_order.py
@@ -47,7 +47,6 @@ SALE_ORDER_STATE = [
 
 
 class SaleOrder(models.Model):
-    _name = 'sale.order'
     _inherit = ['portal.mixin', 'product.catalog.mixin', 'mail.thread', 'mail.activity.mixin', 'utm.mixin']
     _description = "Sales Order"
     _order = 'date_order desc, id desc'

--- a/addons/sale/models/sale_order_line.py
+++ b/addons/sale/models/sale_order_line.py
@@ -14,8 +14,7 @@ from odoo.tools.translate import _
 
 
 class SaleOrderLine(models.Model):
-    _name = 'sale.order.line'
-    _inherit = 'analytic.mixin'
+    _inherit = ['analytic.mixin']
     _description = "Sales Order Line"
     _rec_names_search = ['name', 'order_id.name']
     _order = 'order_id, sequence, id'

--- a/addons/sale/models/utm_campaign.py
+++ b/addons/sale/models/utm_campaign.py
@@ -3,8 +3,9 @@
 
 from odoo import fields, models
 
+
 class UtmCampaign(models.Model):
-    _inherit = 'utm.campaign'
+    _inherit = ['utm.campaign']
     _description = 'UTM Campaign'
 
     quotation_count = fields.Integer('Quotation Count',
